Project.toml

```diff
@@ -1,9 +1,13 @@
 name = "ChainRulesCore"
 uuid = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
-version = "0.5.1"
+version = "0.5.2"
+
+[deps]
+MuladdMacro = "46d2c3a1-f734-5fdb-9937-b9b9aeba4221"
 
 [compat]
 julia = "^1.0"
+MuladdMacro = "0.2.1"
```

Comment:
I am not thrilled about adding this dependency,
because it brings down MacroTools and DataStructures.
But I think it's worth it.

Reply:
add MacroTools after we went to the effot of removing Casette seems a bit sad. might be worth it, but can we see some benchmarks first?

Reply:
It's definately going to be worth it.